modules/imgproc/src/smooth.cpp

```diff
@@ -3025,28 +3055,31 @@ template <>
 void vlineSmooth1N<uint8_t, ufixedpoint16>(const ufixedpoint16* const * src, const ufixedpoint16* m, int, uint8_t* dst, int len)
 {
     const ufixedpoint16* src0 = src[0];
-    v_uint16x8 v_mul = v_setall_u16(*((uint16_t*)m));
-#if CV_SSE2
-    v_uint16x8 v_1 = v_setall_u16(1);
+    int i = 0;
+#if CV_SIMD
+    const int VECSZ = v_uint16::nlanes;
+    v_uint16 v_mul = vx_setall_u16(*((uint16_t*)m));
+#if CV_SSE2 && !CV_SIMD512 && !CV_SIMD256
+    v_uint16 v_1 = vx_setall_u16(1);
     v_mul += v_mul;
 #endif
-    int i = 0;
-    for (; i <= len - 16; i += 16)
+    for (; i <= len - 2*VECSZ; i += 2*VECSZ)
     {
-        v_uint16x8 v_src0 = v_load((uint16_t*)src0 + i);
-        v_uint16x8 v_src1 = v_load((uint16_t*)src0 + i + 8);
-        v_uint8x16 v_res;
-#if CV_SSE2
+        v_uint16 v_src0 = vx_load((uint16_t*)src0 + i);
+        v_uint16 v_src1 = vx_load((uint16_t*)src0 + i + VECSZ);
+        v_uint8 v_res;
+#if CV_SSE2 && !CV_SIMD512 && !CV_SIMD256
         v_res.val = _mm_packus_epi16(_mm_srli_epi16(_mm_add_epi16(v_1.val, _mm_mulhi_epu16(v_src0.val, v_mul.val)),1),
                                      _mm_srli_epi16(_mm_add_epi16(v_1.val, _mm_mulhi_epu16(v_src1.val, v_mul.val)),1));
```

Comment:
Does we still need this optimization branch?

BTW, There is [zero coverage](http://pullrequest.opencv.org/buildbot/export/opencv_releases/master_coverage-lin64-debug/20180727-005450_10527/coverage_html/opencv/modules/imgproc/src/smooth.cpp.gcov.html) for line.